apps/theming-designer/src/components/ThemingDesigner.tsx

```diff
@@ -0,0 +1,202 @@
+import * as React from 'react';
+import { FabricPalette } from './FabricPalette';
+import { AccessibilityChecker } from './AccessibilityChecker';
+import { SemanticSlots } from './SemanticSlots';
+import { Header } from './Header';
+import { Samples } from './Samples';
+import { mergeStyles } from '@uifabric/merge-styles';
+import {
+  Stack,
+  loadTheme,
+  ITheme,
+  getTheme,
+  BaseComponent,
+  IColor,
+  getColorFromString,
+  IconButton,
+  ThemeProvider,
+  createTheme
+} from '../../../../packages/office-ui-fabric-react/lib/index';
+import { ThemeDesignerColorPicker } from './ThemeDesignerColorPicker';
+
+import { ThemeGenerator, themeRulesStandardCreator, BaseSlots, IThemeRules } from 'office-ui-fabric-react/lib/ThemeGenerator';
+import { isDark } from 'office-ui-fabric-react/lib/utilities/color/shades';
+
+export interface IThemingDesignerState {
+  primaryColor: IColor;
+  textColor: IColor;
+  backgroundColor: IColor;
+  theme?: ITheme;
+  themeRules?: IThemeRules;
+}
+
+const appClassName = mergeStyles({
+  display: 'flex'
+});
+
+const sideBarClassName = mergeStyles({
+  flex: '0 0 400px',
+  background: '#fff'
+});
+
+const mainCardsClassName = mergeStyles({
+  flexDirection: 'column',
+  background: '#fff',
+  flex: 1,
+  margin: 0,
+  padding: 0
+});
+
+export class ThemingDesigner extends BaseComponent<{}, IThemingDesignerState> {
+  constructor(props: any) {
+    super(props);
+
+    const themeRules = themeRulesStandardCreator();
+    ThemeGenerator.insureSlots(themeRules, isDark(themeRules[BaseSlots[BaseSlots.backgroundColor]].color!));
```

Comment:
I realize this was pre-existing, but should be `ensureSlots`. Is it a huge impact to change it?

Reply:
done.